### ISSUE TITLE
Connection \"default\" was not found.

### ISSUE THREAD

**Issue by vickylance** [typeorm/typeorm#5362]
**Issue type:**

[ ] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[x] `postgres`
[ ] `cockroachdb`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[x] `0.2.22`

**Steps to reproduce or a small repository showing the problem:**

I have my `createConnection` file as follows.

```ts
import { getConnectionOptions, createConnection, Connection } from "typeorm";

export let connection: Connection | undefined = undefined;

export const createTypeORMConnection = async () => {
  console.log(process.env.NODE_ENV!);
  const connectionOptions = await getConnectionOptions(process.env.NODE_ENV!);
  console.log(connectionOptions);
  connection = await createConnection(connectionOptions);
  return connection;
};
```

The console logs are printing fine.

And my Resolver I am using the connection object exported here and using the save method.

```ts
console.log("env", process.env.NODE_ENV, connection!.name);
const user = await connection!
  .getRepository<User>(User)
  .create({
    firstName,
    lastName,
    email,
    password: hashedPassword
  })
  .save();
console.log(`register - Create User: ${user} -> Done`);
```

but before hitting the `register - create user` console log in the `save` method itself I am getting Connection default was not found.

**Comment by Cali93** [comment]
@vickylance Your ormConfig file should be something like this:
`{
      name: 'default',
      type: 'postgres',
      ...
}`
More infos: https://github.com/typeorm/typeorm/blob/master/docs/connection.md

**Comment by vickylance** [comment]
@Cali93 But what if I want to use different names for different environments? Its an array of objects and only one of that object will be used based on Node_ENV

**Comment by vickylance** [comment]
Sorry accidentally closed

**Comment by Cali93** [comment]
@vickylance In that can I would use something like [config](https://www.npmjs.com/package/config). It will load the right config based on your NODE_ENV, so you'll always have the default connection with the proper config params.

**Comment by vickylance** [comment]
@Cali93 Ok it seems to work fine. But I am still curious as to why the library looks for a connection named default when I have literally created and exported the connection and I am executing methods/queries on that exported connection. Wouldn't it make sense to just use the connection which I created instead of looking out for an elusive "default" connection?

**Comment by Cali93** [comment]
@vickylance So that we can use several connections in an architecture where we would have a DB connection per entity let's say. Can you please close the issue ? Thanks in advance.

**Comment by aounleonardo** [comment]
@Cali93 I'm sorry I don't understand your explanation :/ If I am already exporting an array of 3 connectionOptions ("test", "dev", "prod"), why do I have to still have a "default" one exported if I'm never going to call getConnection without an argument?

**Comment by giacomorebonato** [comment]
Same situation here:
```js
const dbOptions = await getConnectionOptions(NODE_ENV) // production | development`
```
Is taking the right db option... and then it fails on a `default` connection that is not mentioned anywhere.

**Comment by giacomorebonato** [comment]
Worked for me. Thanks.

**Comment by thiagovasconcellos** [comment]
Guys, I was facing the same problem. Took me a while to realize that I was calling a repository outside the method:

`import { getCustomRepository } from 'typeorm';
import CustomerRepository from '../repositories/CustomerRepository';
import { Customer } from '../entities/Customer';

interface IRequest {
  name: string;
  contactPhoneId: string;
  isValid: boolean;
}
**const customerRepository = getCustomRepository(CustomerRepository);**

class CreateCustomerService {
  public async execute({
    name,
    contactPhoneId,
    isValid,
  }: IRequest): Promise<Customer> {
    const customer = customerRepository.create({
      name,
      contactPhoneId,
      isValid,
    });
    await customerRepository.save(customer);
    return customer;
  }
}

export default CreateCustomerService;`

This method was called inside the routes of express and tries to connect to the repository before the connection even exists.

**Comment by sonishubham65** [comment]
constructor(
    @InjectConnection(connectionName) private connection: Connection,
    private jwtService: JwtService,
    private settingService: SettingService,
  ) {
    this.verificationEntityRepositry = getCustomRepository(
      VerificationEntityRepository,
      this.connection.name,
    );
    this.authEmailHelper = new AuthEmailHelper(this.settingService);
    this.userHelper = new UserHelper(this.settingService);
  }
  
  
  
  const module: TestingModule = await Test.createTestingModule({
      providers: [
        VerificationService,
        {
          provide: getConnectionToken(connectionName),
          useValue: {},
        },
          provide: getCustomRepositoryToken(VerificationEntityRepository),
      ],
    }).compile();
